src/Workspaces/CSharp/Portable/Simplification/CSharpSimplificationService.cs

```diff
@@ -184,5 +185,51 @@ protected override void GetUnusedNamespaceImports(SemanticModel model, HashSet<S
                 }
             }
         }
+
+        private static bool IsTupleInDeconstruction(SyntaxNode tuple)
+        {
+            Contract.Assert(tuple.IsKind(SyntaxKind.TupleExpression));
+            var parent = tuple.Parent;
+            do
+            {
+                if (parent.IsKind(SyntaxKind.SimpleAssignmentExpression))
+                {
+                    return true;
+                }
+            }
+            while ((tuple = parent.Parent).IsKind(SyntaxKind.TupleExpression)
+                && (parent = parent.Parent.Parent) != null);
+
+            return false;
+        }
+
+        // This code is copied from the compiler
```

Comment:
Let's have a larger discussion. @CyrusNajmabadi and I talked, and I don't think comfortable with the fragility of copying code here. I the IDE *and* the compiler needs this method, then it should be a public API because somebody else will almost certainly need it too. :smile: